filecoin/functions/handle-filecoin-submit-message.js

```diff
@@ -34,9 +36,26 @@ async function handleFilecoinSubmitMessage (sqsEvent) {
   })
 
   // create context
-  const { pieceTableName } = getEnv()
+  const {
+    pieceTableName,
+    s3BucketName,
+    r2BucketName,
+    r2BucketEndpoint,
+    r2BucketAccessKeyId,
+    r2BucketSecretAccessKey
+  } = getEnv()
   const context = {
-    pieceStore: createPieceTable(AWS_REGION, pieceTableName)
+    pieceStore: createPieceTable(AWS_REGION, pieceTableName),
+    dataStore: composeDataStoresWithOrderedStream(
```

Comment:
we want to read from S3 first, fallback to R2 if not there (prep for new world)

Reply:
Nit: this would read more naturally if it was something like

```js
cons s3 = createDataStore(AWS_REGION, s3BucketName)
const r2 = createDataStore(R2_REGION, r2BucketName, {
    endpoint: r2BucketEndpoint,
    credentials: {
      accessKeyId: r2BucketAccessKeyId,
      secretAccessKey: r2BucketSecretAccessKey,
    },
})

const store = S3.or(r2)
```

Reply:
good call, we actually already use this pattern on `upload-api` https://github.com/web3-storage/w3infra/blob/main/upload-api/functions/ucan-invocation-router.js#L185C52-L186C1 and that is why I kept. But probably we could get more declarative on a follow up